src/ray/protobuf/job_agent.proto

```diff
@@ -0,0 +1,41 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package ray.rpc;
+
+import "src/ray/protobuf/agent_manager.proto";
+
+message InitializeJobEnvRequest {
+  // The job id.
+  bytes job_id = 1;
+  // The job description in JSON.
+  string job_description = 2;
+}
+
+message InitializeJobEnvReply {
+  // The status of job env initialization.
+  AgentRpcStatus status = 1;
+  // The error message in InitializeJobEnv.
+  string error_message = 2;
+  // The driver pid of job.
+  int32 driver_pid = 3;
+}
+
+// Service for communicating with the job_agent.py process on a remote node.
+service JobAgentService {
+  // Initialize job env.
+  rpc InitializeJobEnv(InitializeJobEnvRequest) returns (InitializeJobEnvReply);
+}
```

Comment:
Is there some way to determine when a job has finished?

Reply:
The job status can be queried by job id from `GET /jobs/{job id}`. In the following drop job PR, raylet will have some logic to detect driver exit.